validator/project_validator.go

```diff
@@ -989,9 +989,11 @@ func validateTimeoutLimits(_ context.Context, settings *evergreen.Settings, proj
 		for _, task := range project.Tasks {
 			if task.ExecTimeoutSecs > settings.TaskLimits.MaxExecTimeoutSecs {
 				errs = append(errs, ValidationError{
-					Message: fmt.Sprintf("task '%s' exec timeout (%d) exceeds maximum limit (%d)", task.Name, task.ExecTimeoutSecs, settings.TaskLimits.MaxExecTimeoutSecs),
-					Level:   Error,
+					Message: fmt.Sprintf("task '%s' exec timeout (%d) is too high and will be set to maximum limit (%d)", task.Name, task.ExecTimeoutSecs, settings.TaskLimits.MaxExecTimeoutSecs),
+					Level:   Warning,
 				})
+
+				task.ExecTimeoutSecs = settings.TaskLimits.MaxExecTimeoutSecs
```

Comment:
I could see overriding the timeout in the local project being a little fragile because setting it here relies on the fact that this is a pointer (and none of the structs are ever copied anywhere), and that this pointer's modified value eventually gets saved backed into the DB as the parser project (outside of the validation logic). I believe as this is, you would have to modify the parser project, not the project, because the parser project is what's saved back to the DB. Generally, playing with pointer values like this and relying on the external caller to actually save the value as-is back to the DB is vulnerable to bugs because the pointed-to value can be copied. Usually, if we want a value to persist in the DB, we usually save something back to the DB immediately to update the state.

Instead of setting the max on the parser project itself, could we do something like enforce the max exec timeout [in the agent](https://github.com/evergreen-ci/evergreen/blob/35915149cae377a867e6da0bdd3197e42a7e8e30/agent/task_context.go#L288-L301) instead?

Reply:
I think that makes sense.

It would still be more clean to enforce this in the validate so I created DEVPROD-11204 and wrote todos to undo this change once we can enforce it

Reply:
Would it be bad for it to just stay a warning for good rather than increase it to an error? While I don't imagine we'd adjust it that often, I kind of feel like putting a cap on it and warning that it might not give them the full duration is possibly good enough. Keeping it a warning also means we could adjust the limit in the future if we wanted to, and it won't interfere with any YAMLs.

Reply:
I previously discussed it with @ablack12 and she said that it would probably be cleaner for our validation process for it to eventually be an error so users can't add tasks with super long limits later and question why it wasn't honored because they didn't read the warning